src/index.ts

```diff
@@ -1,11 +1,202 @@
-import { Handler, FirehoseTransformationEvent, FirehoseTransformationEventRecord } from 'aws-lambda'
+import { Handler, FirehoseTransformationEvent, FirehoseTransformationResult, FirehoseTransformationResultRecord } from 'aws-lambda'
 
-export const handler: Handler = async (event: FirehoseTransformationEvent) => {
-  const output = event.records.map((record: FirehoseTransformationEventRecord) => ({
-    recordId: record.recordId,
-    result: 'Ok',
-    data: record.data,
-  }))
+type SplunkRecord = {
+  host: string
+  source: string
+  sourcetype: string
+  index: string
+  event: string
+  fields: {
+    account: string
+    environment: string
+  }
+}
+
+type CloudWatchRecordData = {
+  owner: string
+  logGroup: string
+  logStream: string
+  subscriptionFilters: []
+  messageType: string
+  logEvents: [
+    {
+      id: string
+      timestamp: string
+      message: string
+    }
+  ]
+}
+
+type S3LogRecord = {
+  SourceFile: {
+    S3Bucket: string,
+    S3Key: string
+  },
+  S3Bucket?: string,
+  ALB?: string,
+  AWSAccountID: string,
+  AWSAccountName: string,
+  Logs: string[]
+}
+
+type EnvVars = {
+  environment: string,
+  account: string
+}
+
+function sourceTypeFromLogGroup(logGroup: string): string {
+  const logType = logGroup.split('_')[1]
+  switch(logType) {
+    case 'app':
+      return 'ST004:application_json'
+    default:
+      throw Error(`Unknown source type for log type of "${logType}"`)
+  }
+}
+
+function sourceFromLogGroup(logGroup: string): string {
+  const source = logGroup.split('_').pop()
+  if (!source) {
+    throw new Error(`Unable to extract source from log group ${logGroup}`)
+  }
+  return source
+}
+
+function indexFromLogGroup(logGroup: string): string {
+  const logType = logGroup.split('_')[1]
+  switch(logType) {
+    case 'app':
+      return 'pay_application'
```

Comment:
(minor) would be useful to use env variables for index so we can override for e2e test

Reply:
I'd like to wait until we've made a decision on that before adding it in please.